diff --git a/l10n_es_dua_sii/models/account_invoice.py b/l10n_es_dua_sii/models/account_invoice.py
index 497fa53..a4d7805 100644
--- a/l10n_es_dua_sii/models/account_invoice.py
+++ b/l10n_es_dua_sii/models/account_invoice.py
@@ -36,7 +36,7 @@ class AccountInvoice(models.Model):
         taxes = self._get_sii_taxes_map(['DUA'])
         for invoice in self:
             dua_fiscal_position = self._get_dua_fiscal_position(
-                self.company_id)
+                invoice.company_id)
             invoice.sii_dua_invoice = \
                 invoice.fiscal_position_id == dua_fiscal_position and \
                 invoice.tax_line_ids.filtered(lambda x: x.tax_id in taxes)
