diff --git a/openerp/osv/fields.py b/openerp/osv/fields.py
index 0eae418..5b23c73 100644
--- a/openerp/osv/fields.py
+++ b/openerp/osv/fields.py
@@ -677,8 +677,10 @@ class one2many(_column):
                 field_table = obj.pool[field_model]._table
                 cr.execute("select 1 from {0} where id=%s and {1}=%s".format(field_table, self._fields_id), (act[1], id))
                 if not cr.fetchone():
-                    # Must use write() to recompute parent_store structure if needed and check access rules
-                    obj.write(cr, user, [act[1]], {self._fields_id:id}, context=context or {})
+                    cr.execute("select 1 from {0} where id=%s".format(field_table, self._fields_id), (act[1],))
+                    if cr.fetchone():
+                        # Must use write() to recompute parent_store structure if needed and check access rules
+                        obj.write(cr, user, [act[1]], {self._fields_id:id}, context=context or {})
             elif act[0] == 5:
                 reverse_rel = obj._all_columns.get(self._fields_id)
                 assert reverse_rel, 'Trying to unlink the content of a o2m but the pointed model does not have a m2o'
