diff --git a/l10n_it_ricevute_bancarie/models/riba.py b/l10n_it_ricevute_bancarie/models/riba.py
index 95c13ef..2968c9a 100644
--- a/l10n_it_ricevute_bancarie/models/riba.py
+++ b/l10n_it_ricevute_bancarie/models/riba.py
@@ -263,7 +263,7 @@ class RibaListLine(models.Model):
         'account.account', string='Acceptance Account')
     bank_id = fields.Many2one('res.partner.bank', string='Debitor Bank')
     iban = fields.Char(
-        related='bank_id.acc_number', string='IBAN', store=False,
+        compute='_get_partner_iban', string='IBAN', store=False,
         readonly=True)
     distinta_id = fields.Many2one(
         'riba.distinta', string='List', required=True, ondelete='cascade')
@@ -406,6 +406,14 @@ class RibaListLine(models.Model):
             to_be_settled.reconcile()
             settlement_move.post()

+    @api.multi
+    def _get_partner_iban(self):
+        for line in self:
+            if line.partner_id.bank_ids:
+                mandates = line.partner_id.bank_ids.filtered(lambda i: 'cancel' not in i.mandate_ids.mapped('state'))
+                if mandates:
+                    line.iban = mandates[0].acc_number
+

 class RibaListMoveLine(models.Model):
 
