diff --git a/openerp/models.py b/openerp/models.py
index 6e09504..890959a 100644
--- a/openerp/models.py
+++ b/openerp/models.py
@@ -1807,6 +1806,9 @@ class BaseModel(object):
                     defaults[dv] = [(0, 0, x) for x in defaults[dv]]
             defaults.update(values)
             values = defaults
+        if str(self) == 'stock.move':
+            values.pop('product_qty', False)
+        # discard magic fields
         return values

     def clear_caches(self):
