diff --git a/addons/stock/stock.py b/addons/stock/stock.py
index 8da55a8..97ac74e 100644
--- a/addons/stock/stock.py
+++ b/addons/stock/stock.py
@@ -1609,7 +1609,8 @@ class stock_move(osv.osv):
             for `product_qty`, where the same write should set the `product_uom_qty` field instead, in order to
             detect errors.
         """
-        raise osv.except_osv(_('Programming Error!'), _('The requested operation cannot be processed because of a programming error setting the `product_qty` field instead of the `product_uom_qty`.'))
+        if value > 0:
+            raise osv.except_osv(_('Programming Error!'), _('The requested operation cannot be processed because of a programming error setting the `product_qty` field instead of the `product_uom_qty`.'))
 
     _columns = {
         'name': fields.char('Description', required=True, select=True),
