diff --git a/addons/web/static/src/js/view_list.js b/addons/web/static/src/js/view_list.js
index 16d0310..fe978c2 100644
--- a/addons/web/static/src/js/view_list.js
+++ b/addons/web/static/src/js/view_list.js
@@ -1575,13 +1575,14 @@ instance.web.ListView.Groups = instance.web.Class.extend( /** @lends instance.we
                     // if drag to 1st row (to = 0), start sequencing from 0
                     // (exclusive lower bound)
                     seq = to ? list.records.at(to - 1).get(seqname) : 0;
-                var fct = function (dataset, id, seq) {
+                var fct = function (dataset, id, seq, options) {
                     $.async_when().done(function () {
                         var attrs = {};
                         attrs[seqname] = seq;
-                        dataset.write(id, attrs);
+                        dataset.write(id, attrs, options);
                     });
                 };
+                first=true;
                 while (++seq, (record = list.records.at(index++))) {
                     // write are independent from one another, so we can just
                     // launch them all at the same time and we don't really
@@ -1589,7 +1590,10 @@ instance.web.ListView.Groups = instance.web.Class.extend( /** @lends instance.we
                     // FIXME: breaks on o2ms (e.g. Accounting > Financial
                     //        Accounting > Taxes > Taxes, child tax accounts)
                     //        when synchronous (without setTimeout)
-                    fct(dataset, record.get('id'), seq);
+                    options = {'context':{}}
+                    options.context = {'first':first}
+                    first=false;
+                    fct(dataset, record.get('id'), seq, options);
                     record.set(seqname, seq);
                 }
             }
